pkg/entity/flag.go

```diff
@@ -95,3 +86,29 @@ func (qs FlagQuerySet) DescriptionLike(description string) FlagQuerySet {
 		fmt.Sprintf("%%%s%%", strings.ToLower(description)),
 	))
 }
+
+// CreateFlagKey creates the key based on the given key
+func CreateFlagKey(key string) (string, error) {
+	if key == "" {
+		key = util.NewSecureRandomKey()
+	} else {
+		ok, reason := util.IsSafeKey(key)
+		if !ok {
+			return "", fmt.Errorf("cannot create flag due to invalid key. reason: %s", reason)
+		}
+	}
+	return key, nil
+}
+
+// CreateFlagEntityType creates the FlagEntityType if not exists
+func CreateFlagEntityType(db *gorm.DB, key string) error {
+	ok, reason := util.IsSafeKey(key)
+	if !ok && key != "" {
```

Comment:
Instead of just checking if the key is valid, I wonder if we can add an ENV variable to limit the possible values of entityTypes here.

Reply:
I'm not sure that an ENV variable will add more standardization/is less prone to error than adding in the flagr UI - seems like just one more annoying step. However, this one field does have the potential to affect analytics in real-time and could make things very wonky

Reply:
Maybe we can fork the open source UI and create a private one just for Checkr.

Reply:
As discussed, we will make it a UI env in the next pr